### PR TITLE
refactor: store `is_export_star` info in `ImportRecord` meta directly

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
### Description
1. since they just did the same thing and even with vec `star_exports` we still need to get the original `ImportRec` by index, and then do some stuff.
2. this refactor also save some memory

### CODE REVIEW

File: crates/rolldown/src/stages/link_stage/wrapping.rs [hunk 1]
```diff
@@ -103,7 +103,7 @@ impl LinkStage<'_> {
         );
       }
 
-      if !module.star_exports.is_empty() {
+      if module.has_star_export() {
```
Comment (hyf0) [inline crates/rolldown/src/stages/link_stage/wrapping.rs:106]: Most of modules doesn't have `star_export`, this cause a `o(1)` check method become `o(n)` check which is called on every modules.

File: crates/rolldown/src/stages/link_stage/mod.rs [hunk 2]
```diff
@@ -235,24 +235,27 @@ impl<'a> LinkStage<'a> {
...
-                .entry(rec.resolved_module)
-                .or_insert_with(|| {
-                  // Created `SymbolRef` is only join the de-conflict process to avoid conflict with other symbols.
-                  self.symbols.create_facade_root_symbol_ref(
-                    importer.idx,
-                    legitimize_identifier_name(&ext.name).into_owned().into(),
-                  )
-                });
+        importer
+          .import_records
```
Comment (hyf0) [inline crates/rolldown/src/stages/link_stage/mod.rs:242]: The same as https://github.com/rolldown/rolldown/pull/2497/files#r1814691768.

File: crates/rolldown/src/stages/link_stage/mod.rs [hunk 3]
```diff
@@ -300,7 +303,7 @@ impl<'a> LinkStage<'a> {
               let importee_linking_info = &self.metas[importee.idx];
               match rec.kind {
                 ImportKind::Import => {
-                  let is_reexport_all = importer.star_exports.contains(rec_id);
+                  let is_reexport_all = rec.meta.contains(ImportRecordMeta::IS_EXPORT_START);
```
Comment (hyf0) [inline crates/rolldown/src/stages/link_stage/mod.rs:309]: You could record this meta to get o(1) check here. However, it's better to keep the `star_exports` field for other parts.

File: crates/rolldown/src/stages/link_stage/mod.rs (hunk 2 above)
Comment (hyf0) [reply crates/rolldown/src/stages/link_stage/mod.rs:242]: This one is still slower though it have o(1) check `has_star_export`. Anyway, probably should investigate it when we have profiling data.

**Review by hyf0 — APPROVED** [review]

File: crates/rolldown/src/stages/link_stage/mod.rs (hunk 2 above)
Comment (IWANABETHATGUY) [reply crates/rolldown/src/stages/link_stage/mod.rs:242]: Yeah, but it needs to meet `is_entry` so only a few module will be executed.

**Pull request merged**